### PR TITLE
Gm pricing implement UI

### PR DESCRIPTION
**Pull request by thuynguyen0807**

### CODE REVIEW

**Comment by vqminh** [comment]
It's better to have a description for the PR, Thuy.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+pricing-backend/node_modules
+pricing-backend/dist
+pricing-app/node_modules
```
Comment (vqminh) [inline .gitignore]: can just do node_modules, to exclude them all

File: README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Pricing app
+
+## I. Setup and run
+
+### 1. Backend
+
+```bash
+cd pricing-backend
+npm install
+npm start
```
Comment (vqminh) [inline README.md:10]: Good job having an instruction. In nextjs you can also have server rendering and don't require a separated backend right?

File: pricing-app/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Getting Started
+
```
Comment (vqminh) [inline pricing-app/README.md:4]: Very good documentation.

File: pricing-app/src/app/admin/orders/orderItem.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import { Item, Options } from "@/types/type";
+import { FC } from "react";
+
+type Props = {
+  id: string;
+  createdAt: string;
+  updatedAt: string;
+  cost: number;
+  status: string;
+  onHandleUpdateOrder: (id: string) => void;
```
Comment (vqminh) [inline pricing-app/src/app/admin/orders/orderItem.tsx]: onOrderUpdated is better

File: pricing-app/src/app/admin/orders/orderItem.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  updatedAt,
+  cost,
+  status,
+  onHandleUpdateOrder,
+  onChangeStatusOrder,
+}: Props) => {
+  return (
+    <div className="mb-4">
+      <div className="flex w-full justify-between" onClick={() => {}}>
+        <div className="">{`$${cost}`}</div>
```
Comment (vqminh) [inline pricing-app/src/app/admin/orders/orderItem.tsx:28]: is this necessary?

File: pricing-app/src/app/admin/orders/orderItem.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  id: string;
+  createdAt: string;
+  updatedAt: string;
+  cost: number;
+  status: string;
+  onHandleUpdateOrder: (id: string) => void;
+  onChangeStatusOrder: (value: Options) => void;
+};
+
+const OrderItem: FC<Props> = ({
```
Comment (vqminh) [inline pricing-app/src/app/admin/orders/orderItem.tsx]: The naming here is a bit confusing, may be OrderItemView: FC<{orderItem: OrderItem, onOrderUpdated:....>

File: pricing-app/src/app/admin/orders/page.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+"use client";
+import { updateData } from "@/app/utils";
+import { FC, useEffect, useRef, useState } from "react";
+import OrderItem from "./orderItem";
+import { Options, Order } from "@/types/type";
+import moment from "moment";
+
+const OrderPage: FC = () => {
+  const [items, setItems] = useState<Order[]>([]);
```
Comment (vqminh) [inline pricing-app/src/app/admin/orders/page.tsx:9]: This is a bit old school, think about how to separated the view from the loading logic, something like: useOrder() 
https://www.freecodecamp.org/news/introduction-to-react-hooks/ ==> look at the custom hook

File: pricing-app/src/app/order/OrderBreakfastItem.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  onBreakfastItemAdded: (breakfast: Item) => void;
+};
+
+const OrderBreakfastItem: FC<Props> = ({
+  type,
+  toppingOptions,
+  onBreakfastItemAdded,
+}: Props) => {
+  const [breakfastItem, setBreakfastItem] = useState<Item>(defaultBreakfastItem);
+  const calculateCost = (): number => {
```
Comment (vqminh) [inline pricing-app/src/app/order/OrderBreakfastItem.tsx]: Can we separate this logic out? ideally, each component should have a small single focus purpose.

File: pricing-app/src/app/order/OrderCoffeeItem.tsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+};
+
+const OrderCoffeeItem: FC<Props> = ({
+  type,
+  toppingOptions,
+  sizeOptions,
+  milkOptions,
+  onCoffeeItemAdded,
+}: Props) => {
+  const calculatePrice3 = (coffee: Item): number => {
```
Comment (vqminh) [inline pricing-app/src/app/order/OrderCoffeeItem.tsx]: Should separated this out. Ideally, each component should tell a simple, straightforward story.

File: README.md (hunk 2 above)
Comment (thuynguyen0807) [reply README.md:10]: Yes. I have already setup a separate backend and implement it, after that, I know nextjs also have server rendering. I will try to set up that way. Thank you for your suggestion!

File: pricing-app/README.md (hunk 3 above)
Comment (thuynguyen0807) [reply pricing-app/README.md:4]: Viet Anh helped me to do it.

File: pricing-app/src/app/admin/orders/orderItem.tsx (hunk 5 above)
Comment (thuynguyen0807) [reply pricing-app/src/app/admin/orders/orderItem.tsx:28]: Yes, I want to show the cost of each order.

File: README.md (hunk 2 above)
Comment (thuynguyen0807) [reply README.md:10]: Yes. I have already setup a separate backend and implement it, after that, I know nextjs also have server rendering. I will try to set up that way. Thank you for your suggestion!

File: pricing-app/src/app/admin/orders/page.tsx (hunk 7 above)
Comment (thuynguyen0807) [reply pricing-app/src/app/admin/orders/page.tsx:9]: I will improve it in another PR.

File: pricing-app/src/app/order/OrderBreakfastItem.tsx (hunk 8 above)
Comment (thuynguyen0807) [reply pricing-app/src/app/order/OrderBreakfastItem.tsx]: Yes, I will separate it.

**Comment by thuynguyen0807** [comment]
Hi anh @vqminh, thank you so much for spending time to review my PR. I reworked it and took notes to improve my code. Thank you so much again!

**Pull request merged**